### PR TITLE
BDOG-146 Clean up deprecation warnings

### PR DESCRIPTION
**Pull request by chotai**

### CODE REVIEW

File: app/uk/gov/hmrc/leakdetection/persistence/ReportsRepository.scala [hunk 1]
```diff
@@ -101,25 +102,35 @@ class ReportsRepository @Inject()(reactiveMongoComponent: ReactiveMongoComponent
...
+      key = "repoName",
+      query = Some(hasUnresolvedErrorsSelector),
+      readConcern = ReadConcern.Majority,
+      collation = None).map(_.sorted)
 
   def howManyUnresolved(): Future[Int] =
-    collection.count(Some(Json.obj("inspectionResults" -> Json.obj("$gt" -> JsArray()))))
+    collection.count(
+      selector = Some(Json.obj("inspectionResults" -> Json.obj("$gt" -> JsArray()))),
+      limit = None,
```
Comment (IanMcShane) [inline app/uk/gov/hmrc/leakdetection/persistence/ReportsRepository.scala]: to check, these are all the defaults for count which would be what was done before?

File: app/uk/gov/hmrc/leakdetection/persistence/ReportsRepository.scala [hunk 2]
```diff
@@ -101,25 +102,35 @@ class ReportsRepository @Inject()(reactiveMongoComponent: ReactiveMongoComponent
   def getDistinctRepoNames: Future[List[String]] =
     collection
       .distinct[String, List](
-        "repoName",
-        Some(hasUnresolvedErrorsSelector)
-      )
-      .map(_.sorted)
+      key = "repoName",
+      query = Some(hasUnresolvedErrorsSelector),
+      readConcern = ReadConcern.Majority,
```
Comment (IanMcShane) [inline app/uk/gov/hmrc/leakdetection/persistence/ReportsRepository.scala]: Are these the defaults?  Also the Read Concern could that be set on the connection string? Would this override that behaviour?

File: app/uk/gov/hmrc/leakdetection/persistence/ReportsRepository.scala (hunk 1 above)
Comment (chotai) [reply app/uk/gov/hmrc/leakdetection/persistence/ReportsRepository.scala]: A lot of these are the defaults that I've been able to work out from looking at the calls the deprecated version was making. The exception here being `ReadConcern.Majority`, which I'll comment on below.

File: app/uk/gov/hmrc/leakdetection/persistence/ReportsRepository.scala (hunk 2 above)
Comment (chotai) [reply app/uk/gov/hmrc/leakdetection/persistence/ReportsRepository.scala]: On the non deprecated version of `count` the read concern must be set. There are no defaults and I believe you can not set this on the connection itself (this is my interpretation from reading through the documentation).

There are 5 levels of right concerns that you can set. This is the middle option. This is what this level means (from the documentation)

```
The query returns the data that has been acknowledged by a majority of the replica set members. The documents returned by the read operation are durable, even in the event of failure.
```

Below this level the data written can be rolled back so there is no guarantee on the data. Thinking this through, the draw back of majority might be that it takes a bit longer which I didn't think would be a big problem in this service.

What do you think?

**Pull request merged**